persistence.js

```diff
@@ -65,44 +72,46 @@ function execPipeline (that) {
 }
 
 RedisPersistence.prototype.storeRetained = function (packet, cb) {
-  var key = 'retained:' + packet.topic
   if (packet.payload.length === 0) {
-    this._db.del(key, cb)
+    this._db.hdel(retainedKey, packet.topic, cb)
   } else {
-    this._db.set(key, msgpack.encode(packet), cb)
+    this._db.hset(retainedKey, packet.topic, msgpack.encode(packet), cb)
   }
 }
 
-function checkAndSplit (prefix, pattern) {
+function checkAndSplit (pattern) {
   var qlobber = new Qlobber(qlobberOpts)
   qlobber.add(pattern, true)
 
   var instance = syncthrough(splitArray)
 
   instance._qlobber = qlobber
-  instance._prefix = prefix
 
   return instance
 }
 
 function splitArray (keys, enc) {
-  var prefix = this._prefix.length
   for (var i = 0, l = keys.length; i < l; i++) {
-    var key = keys[i].slice(prefix)
+    var key = keys[i]
     if (this._qlobber.match(key).length > 0) {
       this.push(keys[i])
     }
   }
 }
 
 RedisPersistence.prototype.createRetainedStream = function (pattern) {
+  var that = this
   return new MatchStream({
     objectMode: true,
     redis: this._db,
-    match: 'retained:' + pattern.split(retainedRegexp)[0] + '*',
-    count: 1000
-  }).pipe(checkAndSplit('retained:', pattern))
-    .pipe(throughv.obj(this._decodeAndAugment))
+    key: retainedKey
+  }).pipe(checkAndSplit(pattern))
+    .pipe(through.obj(that._getChunk))
+    .pipe(throughv.obj(decodeRetainedPacket))
```

Comment:
in this block, the source does not need to be a stream, as we are just pushing a single value. Originally, all that code was pushing multiple values on a streaming fashion. Right now we are looking at all values at the same time, and then augmenting using streams.